compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -7,106 +7,208 @@
+    return failure();
+
+  const int64_t targetSubgroupSize = target.getPreferredSubgroupSize();
+
+  SmallVector<int64_t, 4> bounds = linalgOp.getStaticLoopRanges();
+  FailureOr<mlir::linalg::ContractionDimensions> contractionDims =
+      mlir::linalg::inferContractionDims(linalgOp);
+  if (failed(contractionDims)) {
+    return failure();
   }
-  int64_t maxVectorSize = flatNumTrips / numThreads;
 
-  while (maxVectorSize % vectorSize != 0) {
-    vectorSize /= 2;
+  if (contractionDims->k.size() < 1 || contractionDims->m.size() < 1 ||
+      contractionDims->n.size() < 1) {
+    return failure();
   }
 
-  SmallVector<int64_t> tileSizes(loopRanges.size(), 0);
-  tileSizes.back() = vectorSize;
-  int64_t residualNumThreads = numThreads / (loopRanges.back() / vectorSize);
-  for (int i = tileSizes.size() - 2, e = 0; i >= e; --i) {
-    if (loopRanges[i] >= residualNumThreads) {
-      tileSizes[i] = loopRanges[i] / residualNumThreads;
-      residualNumThreads = 1;
-      break;
-    }
-    tileSizes[i] = 1;
-    residualNumThreads /= loopRanges[i];
+  // For now we are not being smart and trying to reshape dimensions to allow
+  // for better usage of intrinsics, and instead are tiling all dimensions
+  // except the inner most m, n, and k dimensions to 1.
+  int64_t mDim = contractionDims->m.back();
+  int64_t nDim = contractionDims->n.back();
+  int64_t kDim = contractionDims->k.back();
+
+  // Dynamic dims are expected to be taken care of earlier in the pipeline.
+  if (ShapedType::isDynamic(bounds[mDim]) ||
+      ShapedType::isDynamic(bounds[nDim]) ||
+      ShapedType::isDynamic(bounds[kDim])) {
+    return failure();
   }
-  return tileSizes;
-}
 
-SmallVector<int64_t> deriveLinalgOpThreadTileSizes(linalg::LinalgOp linalgOp,
-                                                   int64_t numThreads) {
-  if (!linalgOp.hasPureTensorSemantics()) {
-    return {};
+  // Bail out on matvec-like cases.
+  if (bounds[mDim] == 1 || bounds[nDim] == 1) {
+    return failure();
   }
-  // TODO: Support multi-result
-  if (linalgOp->getNumResults() != 1) {
-    return {};
+
+  Value lhs = linalgOp.getDpsInputOperand(0)->get();
+  Value rhs = linalgOp.getDpsInputOperand(1)->get();
+  Value init = linalgOp.getDpsInitOperand(0)->get();
+
+  Type lhsElemType = getElementTypeOrSelf(lhs);
+  Type rhsElemType = getElementTypeOrSelf(rhs);
+  Type initElemType = getElementTypeOrSelf(init);
+
+  GPUMatmulShapeType problem{bounds[mDim], bounds[nDim], bounds[kDim],
+                             lhsElemType,  rhsElemType,  initElemType};
+
+  SmallVector<GPUMatmulShapeType> intrinsics;
+  intrinsics.reserve(target.getWgp().getMma().size());
+  for (IREE::GPU::MMAAttr mma : target.getWgp().getMma()) {
+    auto [mSize, nSize, kSize] = mma.getMNKShape();
+    auto [aType, bType, cType] = mma.getABCElementTypes();
+    if (mma.getSubgroupSize() != targetSubgroupSize)
+      continue;
+    intrinsics.emplace_back(mSize, nSize, kSize, aType, bType, cType);
```

Comment:
Currently the TileAndFuse pipeline only supports the F32_16x16x16_F16 intrinsic and I32_16x16x32xI8 intrinsic. I think it should exclude other intrinsic types here to avoid compilation failures.